imap_processing/idex/IDEXPacketParser.py

```diff
@@ -0,0 +1,120 @@
+
+        self.data = datastore
+        self.numevents = evtnum
+    
+    def _log_packet_info(self, evtnum, pkt):
+        print(f"^*****Event header {evtnum}******^")
+        # Extract the 17-22-bit integer (usually 8)
+        self.lspretrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value >> 16) &  0b1111
+        # Extract the next 4-bit integer (usually 8)
+        self.lsposttrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value >> 12) & 0b1111
+        # Extract the next 6 bits integer (usually 32)
+        self.hspretrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value >> 6) & 0b111111
+        # Extract the first 6 bits (usually 32)
+        self.hsposttrigblocks = (pkt.data['IDX__TXHDRBLOCKS'].derived_value) & 0b111111
+        print("HS pre trig sampling blocks: ", self.hspretrigblocks)
+        print("LS pre trig sampling blocks: ", self.lspretrigblocks)
+        print("HS post trig sampling blocks: ", self.hsposttrigblocks)
+        print("LS post trig sampling blocks: ", self.lsposttrigblocks)
+        self.TOFdelay = pkt.data['IDX__TXHDRSAMPDELAY'].raw_value >> 2  # First two bits are padding
+        mask = 0b1111111111
+        self.lgdelay = (self.TOFdelay) & mask
+        self.mgdelay = (self.TOFdelay >> 10) & mask
+        self.hgdelay = (self.TOFdelay >> 20) & mask
+        print(f"High gain delay = {self.hgdelay} samples.")
+        print(f"Mid gain delay = {self.mgdelay} samples.")
+        print(f"Low gain delay = {self.lgdelay} samples.")
+        if(pkt.data['IDX__TXHDRLSTRIGMODE'].derived_value!=0):  # If this was a LS (Target Low Gain) trigger
+            self.Triggerorigin = 'LS' 
+            print("Low sampling trigger mode enabled.")
+        print("Packet trigger mode = ", pkt.data['IDX__TXHDRLGTRIGMODE'].derived_value, pkt.data['IDX__TXHDRMGTRIGMODE'].derived_value, pkt.data['IDX__TXHDRHGTRIGMODE'].derived_value)
+        if(pkt.data['IDX__TXHDRLGTRIGMODE'].derived_value!=0):
+            print("Low gain TOF trigger mode enabled.")
+            self.Triggerorigin = 'LG'
+        if(pkt.data['IDX__TXHDRMGTRIGMODE'].derived_value!=0):
+            print("Mid gain TOF trigger mode enabled.")
+            self.Triggerorigin = 'MG'
+        if(pkt.data['IDX__TXHDRHGTRIGMODE'].derived_value!=0):
+            print("High gain trigger mode enabled.")
+            self.Triggerorigin = 'HG'
+        print(f"AID = {pkt.data['IDX__SCI0AID'].derived_value}")  # Instrument event number
+        print(f"Event number = {pkt.data['IDX__SCI0EVTNUM'].raw_value}")  # Event number out of how many events constitute the file
+        print(f"Rice compression enabled = {bool(pkt.data['IDX__SCI0COMP'].raw_value)}")   
+        compressed = bool(pkt.data['IDX__SCI0COMP'].raw_value)  # If we need to decompress the data
+        print(f"Timestamp = {self.epochs[evtnum-1]} seconds since epoch (Midnight January 1st, 2012)")
+
+    def _parse_hs_waveform(self, waveform_raw: str):
```

Comment:
You aren't accessing `self` at all, so this could be a static method on the class, but again this seems like a helper function within this module so isn't needed on the class explicitly.